jupyterhub/templates/hub/deployment.yaml

```diff
@@ -5,8 +5,8 @@ metadata:
   labels:
     {{- include "jupyterhub.labels" . | nindent 4 }}
 spec:
-  {{- if typeIs "int" .Values.hub.revisionHistoryLimit }}
-  revisionHistoryLimit: {{ .Values.hub.revisionHistoryLimit }}
+  {{- if .Values.hub.revisionHistoryLimit }}
+  revisionHistoryLimit: {{ .Values.hub.revisionHistoryLimit | int }}
```

Comment:
We should be [able to set this to 0 as well](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#clean-up-policy), and that was the point of having the `typeIs` check. We could also change this check to a "not null" check since we have a schema for the helm chart enforcing that makes it either null or int.

Do you have an example reproduction of the bug you look to fix with this change?

Reply:
Good point, forgot that 0 was treated as a false in this case. I've changed the check to check for not-nil.

Setting the following value file will not render the revisionLimitHistory in the chart:
```yaml
hub:
  revisionHistoryLimit: 1
```

This is on `helm version`:
```
version.BuildInfo{Version:"v3.14.2", GitCommit:"c309b6f0ff63856811846ce18f3bdc93d2b4d54b", GitTreeState:"clean", GoVersion:"go1.21.7"}
```